src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTryFinally.cs

```diff
@@ -3758,5 +3759,492 @@ .maxstack  1
 }
 ");
         }
+
+        [Fact]
+        [WorkItem("https://github.com/dotnet/roslyn/issues/67494")]
+        public void WhenWithAlwaysThrowingExpression_01()
+        {
+            var source = """
+class C
+{
+    static bool _throw;
+
+    static void Main()
+    {
+        M();
+
+        _throw = true;
+
+        try 
+        {
+            M();
+        }
+        catch (System.NullReferenceException)
+        {
+            System.Console.Write("Catch");
+        }
+    }
+
+    static void M()
+    {
+        try
+        {
+            M1();
+        }
+        catch when (true ? throw M2() : true)
+        {
+            M3();
+        }
+
+        M4();
+    }
+
+    static void M1()
+    {
+        System.Console.Write("M1");
+        if (_throw) throw null;
+    }
+
+    static System.Exception M2()
+    {
+        System.Console.Write("M2");
+        return new System.NotSupportedException();
+    }
+
+    static void M3()
+    {
+        System.Console.Write("M2");
```

Comment:
nit: The method name in this call doesn't match the containing method. Consider removing the call if it is not used by the test.

Reply:
```suggestion
        System.Console.Write("M3");
```